tests/baselines/reference/tsbuild/commandLine/emitDeclarationOnly-false-on-commandline-with-declaration-and-incremental-discrepancies.js

```diff
@@ -115,127 +115,4 @@ IncrementalBuild:
     ]
   },
   "version": "FakeTSVersion"
-}
-TsBuild info text without affectedFilesPendingEmit:: /src/project2/src/tsconfig.tsbuildinfo.readable.baseline.txt::

```

Comment:
@sheetalkamat As pictured here, apparently the bug where we didn't update the `buildinfo` when we didn't emit but semantic diagnostics may have changed was already captured in a baseline. There's just other stuff here, so it's noisy.

Reply:
This is incorrectly emitting buildInfo because "emitDeclarationsOnly" does not affect semantic diagnostics. The condition in builder to check for this is: 
```ts
 const canCopySemanticDiagnostics = useOldState && oldState!.semanticDiagnosticsPerFile && !!state.semanticDiagnosticsPerFile &&
        !compilerOptionsAffectSemanticDiagnostics(compilerOptions, oldCompilerOptions!);
```

where as you want it to be `compilerOptionsAffectSemanticDiagnostics(compilerOptions, oldCompilerOptions!);` when emitting buildInfo again. 

I think you just want to check if "noCheck" is changed and then mark the buildInfo emit pending.

Reply:
> I think you just want to check if "noCheck" is changed and then mark the buildInfo emit pending.

How is `noCheck` any different than any other flag that reduces the number of errors produced? Like toggling `noImplicitAny` to `false`? I'm not seeing why anything should be `noCheck`-specific here.